calico-vpp-agent/cni/network_vpp.go

```diff
@@ -214,6 +214,16 @@ func (s *Server) AddVppInterface(podSpec *LocalPodSpec, doHostSideConf bool) (sw
 	}
 	s.log.Infof("created tun[%d]", swIfIndex)
 
+	if tun.NumRxQueues > 1 {
+		for i := 0; i < tun.NumRxQueues; i++ {
+			worker := uint32(tun.NumRxQueues) % s.NumVPPWorkers
```

Comment:
This seems to put all the rx queues on the same worker, is this really what we need?

Reply:
Ah yes, good catch, it's `i % Nworkers`

Reply:
Actually `((tun.SwIfIndex * tun.NumRxQueues) + i) % s.NumVppWorkers` could be even better, otherwise you won't use all workers if you have more workers than queues configured. This configuration may be interesting in high-density clusters...

Reply:
Right !